clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceAfterImageConsumerImpl.java

```diff
@@ -164,12 +173,15 @@ public CompletableFuture<Void> seekToEndOfPush(Set<Integer> partitions) {
                 }
               }
               if (endOfPushConsumedPerPartitionMap.values().stream().allMatch(e -> e)) {
+                LOGGER.info("Found EOP messages for all partitions: " + partitions.toString());
                 // We polled all EOP messages, stop polling!
                 break;
               }
             }
           }
+          LOGGER.info("Seeking to EOP for partitions: " + partitions.toString());
           this.seekToCheckpoint(checkpoints).get();
+          LOGGER.info("Seeked to EOP for partitions: " + partitions.toString());
         } catch (InterruptedException | ExecutionException | VeniceCoordinateOutOfRangeException e) {
           throw new VeniceException(
```

Comment:
What will happen if the VT gets deleted before we can reach EOP for all partitions? Will it be caught here and handled gracefully?

Reply:
Probably not.  This will bubble up an exception, but the storeRepository only logs exceptions for callbacks, it doesn't try again.  So what I'll do is I'll add a retry to the top of the callback in the VersionSwapDetector class